### PR TITLE
Fix some typos and clang-tidy warnings

### PR DESCRIPTION
**Pull request by Lucky-Chang**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Common/ZooKeeper/IKeeper.h [hunk 1]
```diff
@@ -80,7 +80,7 @@ enum class Error : int32_t
     ZUNIMPLEMENTED = -6,        /// Operation is unimplemented
     ZOPERATIONTIMEOUT = -7,     /// Operation timeout
     ZBADARGUMENTS = -8,         /// Invalid arguments
-    ZINVALIDSTATE = -9,         /// Invliad zhandle state
+    ZINVALIDSTATE = -9,         /// Invalid handle state
```
Comment (alexey-milovidov) [inline src/Common/ZooKeeper/IKeeper.h:83]: This is not a typo.

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 2]
```diff
@@ -3894,7 +3894,7 @@ void MergeTreeData::movePartitionToVolume(const ASTPtr & partition, const String
         throw Exception("Volume " + name + " does not exists on policy " + getStoragePolicy()->getName(), ErrorCodes::UNKNOWN_DISK);
 
     if (parts.empty())
-        throw Exception("Nothing to move (сheck that the partition exists).", ErrorCodes::NO_SUCH_DATA_PART);
```
Comment (alexey-milovidov) [inline src/Storages/MergeTree/MergeTreeData.cpp:3897]: Wow.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
How did you find these typos?
Also let's add the new checks to `.clang-tidy`.

**Comment by Lucky-Chang** [comment]
> How did you find these typos? Also let's add the new checks to `.clang-tidy`.

my vscode editor has Code Spell Checker plugin and clangd plugin. it seems they are more powerful than the typo-check in ci and clang-tidy check compile

**Pull request merged**